pkg/gui/view_helpers.go

```diff
@@ -15,6 +15,27 @@ func (gui *Gui) resetOrigin(v *gocui.View) error {
 	return v.SetOrigin(0, 0)
 }
 
+// Returns the number of lines that we should read initially from a cmd task so
+// that the view looks good
+func (gui *Gui) linesToReadFromCmdTask(v *gocui.View) int {
+	_, height := v.Size()
+	_, oy := v.Origin()
+
+	// We want to read as many lines initially as necessary to let the
+	// scrollbar go to its minimum height, so that the scrollbar thumb doesn't
+	// change size as you scroll down.
+	minScrollbarHeight := 2
+	linesToRead := height*(height-1)/minScrollbarHeight + oy
+
+	// However, cap it at some arbitrary max limit, so that we don't get
+	// performance problems for huge monitors or tiny font sizes
+	if linesToRead > 5000 {
+		linesToRead = 5000
```

Comment:
On my machine `linesToRead` is 3655 and that causes noticeable lag in rendering content when going from one branch to another (using branches as an example because the content is effectively infinite).

So if we reduce this number to say 1000 the lag is less noticeable but we've reintroduced the problem that this PR tries to solve.

Between performance and accurate scrollbars, I find performance more important. What are your thoughts?

Reply:
I'm thinking we make this an opt-in feature and then see how many people want to turn it on.

Reply:
For me, the accurate scroll bar size is way more important for diffs than for branch logs, because as you say, branch logs are effectively infinite in practice anyway. And it seems that branch logs are slower to generate than diffs, so it might make sense to use different max linesToRead values for different commands. What do you think?

Reply:
I've done some testing locally and I agree: with diffs we should be pretty safe to keep the scrollbar accurate. By the looks of it it's only the branch panels (branches, tags, remote branches) where we would opt for performance.

Reply:
I pushed a commit to do that. It's maybe not the way we want to pass the information (providing the concrete maxLines values at the call site, I mean), but this way I could most easily see what all the call sites are. Maybe we should use a bool instead (what should we call it?) and then decide in `linesToReadFromCmdTask` what the limit should be based on that. Let me know what you prefer, I don't have much of a preference here.

Reply:
I agree with using a bool. Hard to think of a good name, but if we explain what we're doing in a comment it doesn't matter much. I'd call it readEnoughForScrollbar where we set that value to false for branches, remote branches, and tags. Thoughts?

Reply:
Fine with me; please have a look. I decided to go back to the previous behavior if `readEnoughForScrollbar` is false, instead of capping at different limits, which I would have found hard to explain.

Feel free to amend this if you want the comment in a different place or with different wording.